src/test/java/Homework23.java

```diff
@@ -0,0 +1,35 @@
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import pages.HomePage;
+import pages.LoginPage;
+import pages.SongPage;
+
+public class Homework23 extends BaseTest{
+    LoginPage loginPage;
+    HomePage homePage ;
+    SongPage songPage;
+
+    @BeforeClass
+    void login(){
+        loginPage = new LoginPage(driver);
+        homePage = new HomePage(driver);
+        songPage = new SongPage(driver);
+        loginPage.loginCorrectCred();
+    }
+    
+
+    @Test
+    void contextClickOnSong()  {
+        homePage.clickAllSongs();
+        songPage.contextClickSongByName("Reactor").clickPlay();
+       // homePage.checkSongIsPlaying();
+    }
+    
+
+    @Test
+   void mouseHoverTest() {
+        homePage.clickAllSongs().mouseMoveToPlayBtn().checkIfPlayBtnIsVisible();
+    }
+}
```

Comment:
**Use in test methods, the following logical split:   GIVEN – WHEN - THEN**

For example

![Clean_Test](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/217dadac-49ca-4317-b9dc-2e42a92c32b1)